cpp/src/spatial/hausdorff.cu

```diff
@@ -140,35 +129,29 @@ struct hausdorff_functor
             return result;
         }
 
-        auto d_x = cudf::column_device_view::create(x);
-        auto d_y = cudf::column_device_view::create(y);
-        auto d_trajectory_lengths = cudf::column_device_view::create(trajectory_lengths);
-        auto d_space_offsets = rmm::device_vector<cudf::size_type>(trajectory_lengths.size());
+        auto space_offsets = rmm::device_vector<cudf::size_type>(points_per_space.size());
 
         thrust::inclusive_scan(rmm::exec_policy(stream)->on(stream),
-                                                d_trajectory_lengths->begin<cudf::size_type>(),
-                                                d_trajectory_lengths->end<cudf::size_type>(),
-                                                d_space_offsets.begin());
+                               points_per_space.begin<cudf::size_type>(),
+                               points_per_space.end<cudf::size_type>(),
+                               space_offsets.begin());
 
-        auto kernel = kernel_hausdorff<T>;
-
-        int block_x = result->size();
-        int block_y = 1;
-
-        if (result->size() > 65535)
-        {
-            block_y = ceil((float) result->size() / 65535.0);
-            block_x = 65535;
-        }
+        // utilize one block per result (pair of spaces).
+        int max_block_size = 65535;
```

Comment:
I think the max CUDA thread block size is a lot smaller than this, isn't it?  Maybe this should be named "max_num_blocks"?

Reply:
I found mixed information about this when investigating, but ultimately landed on this because it appeared elsewhere in cudf. I agree that “block_size” is misleading. “max_thread_count” makes more sense.

Reply:
That might be the maximum total number of threads, but when I check for the maximum block x-dimension on my Volta, I get 1024...  

```
mharris@canopus:~/rapids/cudf/cpp/build/release$ vim test.cu
mharris@canopus:~/rapids/cudf/cpp/build/release$ nvcc test.cu
mharris@canopus:~/rapids/cudf/cpp/build/release$ ./a.out 
Max x-dimension: 1024
mharris@canopus:~/rapids/cudf/cpp/build/release$ more test.cu
#include <stdio.h>

int main(void) {
    int max_x{0};
    cudaDeviceGetAttribute(&max_x, cudaDevAttrMaxBlockDimX, 0);
    printf("Max x-dimension: %d\n", max_x);
}
```

This really shouldn't be hard coded...